options/options.go

```diff
@@ -452,6 +454,60 @@ func (opts *ToolOptions) ParseArgs(args []string) ([]string, error) {
 	return args, err
 }
 
+// LogSensitiveOptionWarnings logs a warning for any sensitive information (i.e. passwords)
+// that appear on the command line for the --password, --uri and --sslPEMKeyPassword options.
+// This also applies to a connection string that appears as a positional argument.
+func (opts *ToolOptions) LogSensitiveOptionWarnings(args []string) {
+	passwordMsg := "WARNING: On some systems, a password provided directly using " +
+		"--password may be visible to system status programs such as `ps` that may be " +
+		"invoked by other users. Consider omitting the password to provide it via stdin, " +
+		"or using the --config option to specify a configuration file with the password."
+
+	uriMsg := "WARNING: On some systems, a password provided directly in a connection string " +
+		"or using --uri may be visible to system status programs such as `ps` that may be " +
+		"invoked by other users. Consider omitting the password to provide it via stdin, " +
+		"or using the --config option to specify a configuration file with the password."
+
+	sslMsg := "WARNING: On some systems, a password provided directly using --sslPEMKeyPassword " +
+		"may be visible to system status programs such as `ps` that may be invoked by other users. " +
+		"Consider using the --config option to specify a configuration file with the password."
+
+	// Parse the args to see what is set on the command line.
+	extraArgs, err := opts.parser.ParseArgs(args)
+	if err != nil {
+		return
+	}
+
+	// Parse the extraArgs for a positional connection string.
+	if opts.parsePositionalArgsAsURI {
+		_, err = opts.setURIFromPositionalArg(extraArgs)
+		if err != nil {
+			return
+		}
+	}
+
+	// Log a message for --password, if specified.
+	if opts.Auth.Password != "" {
+		log.Logvf(log.Always, passwordMsg)
+		opts.Auth.Password = ""
```

Comment:
could we add a comment somewhere explaining why we're setting these back to the empty string?

Reply:
I am also confusing about this part, if we are setting this field back to empty string, why not other fields in the commandline?